sap/5.2/modules/ROOT/pages/sap-connector-reference.adoc

```diff
@@ -0,0 +1,638 @@
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Transaction Id a| String |Transaction ID value.  |  |
+|===
+
+[[repeatable-in-memory-stream]]
+=== Repeatable In Memory Stream
+
+When streaming in this mode, Mule never uses the disk to buffer the contents. If you exceed the buffer size, the message fails.
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Initial Buffer Size a| Number | The amount of memory to allocate to consume the stream and provide random access to it. If the stream contains more data than can fit into this buffer, then the buffer expands according to *Buffer Size Increment*, with an upper limit of *Max Buffer Size*. |  |
+| Buffer Size Increment a| Number | This is by how much the buffer size expands if it exceeds its initial size. Setting a value of zero or lower means that the buffer should not expand, meaning that a STREAM_MAXIMUM_SIZE_EXCEEDED error is raised when the buffer gets full. |  |
+| Max Buffer Size a| Number | The maximum amount of memory to use. If more than that is used then a STREAM_MAXIMUM_SIZE_EXCEEDED error is raised. A value lower than or equal to zero means no limit. |  |
+| Buffer Unit a| Enumeration, one of:
+
+** BYTE
+** KB
+** MB
+** GB | The unit in which all these attributes are expressed. |  |
+|===
+
+[[repeatable-file-store-stream]]
+=== Repeatable File Store Stream
+
+File store repeatable streams require buffering, and there are different buffering strategies. Mule keeps a portion of contents in memory. If the stream contents are larger than the configured buffer size, Mule backs up the buffer’s content to disk and then clears the memory.
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| In Memory Size a| Number | Defines the maximum memory that the stream should use to keep data in memory. If more than that is consumed content on the disk is buffered. |  |
+| Buffer Unit a| Enumeration, one of:
+
+** BYTE
+** KB
+** MB
+** GB | The unit in which maxInMemorySize is expressed. |  |
+|===
+
+[[RedeliveryPolicy]]
+=== Redelivery Policy
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Max Redelivery Count a| Number | The maximum number of times a message can be redelivered and processed unsuccessfully before triggering process-failed-message. |  |
+| Use Secure Hash a| Boolean | Whether to use a secure hash algorithm to identify a redelivered message. |  |
+| Message Digest Algorithm a| String | The secure hashing algorithm to use. If not set, the default is SHA-256. |  |
+| Id Expression a| String | Defines one or more expressions to use to determine when a message has been redelivered. This property may only be set if useSecureHash is `false`. |  |
+| Object Store a| Object Store | The object store where the redelivery counter for each message is stored. |  |
+|===
+
```

Comment:
please add here this :


[[IDocParseParameterGroup]]
=== I Doc Parse Parameter Group

[cols=".^20%,.^25%,.^30%,.^15%,.^10%", options="header"]
|======================
| Field | Type | Description | Default Value | Required
| Parse With Field Value Checking a| Boolean | Parsing option for enabling the field value checking mechanism in accordance with the respective IDoc field metadata. | false | 
| Parse Ignore Unknown Fields a| Boolean | Parsing option for ignoring unknown fields, e.g. when parsing with older IDoc metadata. | false | 
| Parse Ignore Invalid Char Errors a| Boolean | Parsing option for ignoring character range check errors in accordance with the declared XML specification version.
 If using this option invalid characters within field values are accepted although the XML document to parse may not be well-formed because it potentially contains characters not being in the valid character range. | false | 
| Parse Without Field Data Type Checking a| Boolean | Parsing option for disabling the field data type checking mechanism in accordance with the respective IDoc field metadata.
 Caution: This option should be used with care.
 Sending IDocs with illegal field contents that are violating the field's defined datatype may cause ABAP system dumps and result in unrecognized partial loss of data when later being processed in an AS ABAP system. | false | 
| Parse Refuse Unknown XML Version a| Boolean | Parsing option for refusing XML documents that declare to use an unknown XML specification version, i.e. a potential future XML specification version 1.2 or 2.0. | false | 
| Parse Refuse XML Version10 a| Boolean | Parsing option for refusing XML documents that declare to use the XML specification version 1.0 or do not contain an XML declaration header. | false | 
| Parse Refuse XML Version11 a| Boolean | Parsing option for refusing XML documents that declare to use the XML specification version 1.1. | false | 
| Parse Accept Only Xml Version10 a| Boolean | Parsing option for accepting only XML documents that declare to use the XML specification version 1.0 or do not contain an XML declaration header. | false | 
| Parse Accept Only Xml Version11 a| Boolean | Parsing option for accepting only XML documents that declare to use the XML specification version 1.1. | false | 
| Parse Accept Only Xml Version10 To11 a| Boolean | Parsing option for accepting only XML documents that declare to use the XML specification version 1.0 or 1.1 or do not contain an XML declaration header. | false | 
|======================